Adjust for pandas 1.2.0

Changes from pandas 1.1.5 to 1.2.0 (released yesterday) caused the nightly CI workflow to fail; in particular `test_timeseries.py::test_get[*-wide]`, as below. Investigate and fix.

As well, [pandas 1.2.0 supports only Python 3.7.1 or later](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v1.2.0.html#increased-minimum-version-for-python), whereas [pandas ≥ 1.0.0 and < 1.2.0 supports Python 3.6.1](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v1.0.0.html#increased-minimum-version-for-python). Consequently, this test failure doesn't appear on CI workflows using Python 3.6; only those using Python ≥ 3.7. We can probably continue to support Python 3.6 / pandas < 1.2.0 until we are forced to make changes for newer pandas that happen to break support for this older combination.

https://github.com/iiasa/ixmp/runs/1613037680?check_suite_focus=true#step:14:541
```
__________________________ test_get[Scenario-wide] ____________________________

ts = <ixmp.core.Scenario object at 0x7f726c91bf70>, fmt = 'wide'

    @pytest.mark.parametrize("fmt", ["long", "wide"])
    def test_get(ts, fmt):
        data = DATA[0] if fmt == "long" else wide(DATA[0])
    
        ts.add_timeseries(data)
        ts.commit("")
    
        exp = expected(data, ts)
        args = {}
    
        if fmt == "wide":
            args["iamc"] = True
    
        # Data can be retrieved and has the expected value
        obs = ts.timeseries(**args)
>       assert_frame_equal(exp, obs, check_like=True)

ixmp/tests/core/test_timeseries.py:269: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/pandas/core/indexes/base.py:4666: in sort_values
    _as = nargsort(
/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/pandas/core/sorting.py:367: in nargsort
    return items.argsort(ascending=ascending, kind=kind, na_position=na_position)
/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/pandas/core/arrays/base.py:586: in argsort
    return nargsort(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

items = array(['model', 'scenario', 'region', 'variable', 'unit', 2010, 2020],
      dtype=object)
kind = 'quicksort', ascending = True, na_position = 'last', key = None
mask = array([False, False, False, False, False, False, False])

    def nargsort(
        items,
        kind: str = "quicksort",
        ascending: bool = True,
        na_position: str = "last",
        key: Optional[Callable] = None,
        mask: Optional[np.ndarray] = None,
    ):

[…]

>       indexer = non_nan_idx[non_nans.argsort(kind=kind)]
E       TypeError: '<' not supported between instances of 'int' and 'str'

/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/pandas/core/sorting.py:379: TypeError
```

--- a/RELEASE_NOTES.rst
+++ b/RELEASE_NOTES.rst
@@ -4,6 +4,7 @@ Next release
 All changes
 -----------
 
+- :pull:`391`: Adjust test suite for pandas v1.2.0.
 - :pull:`374`: Raise clearer exceptions from :meth:`.add_par` for incorrect parameters; silently handle empty data.
 - :pull:`389`: Depend on :mod:`openpyxl` instead of :mod:`xlrd` and :mod:`xlsxwriter` for Excel I/O; :mod:`xlrd` versions 2.0.0 and later do not support :file:`.xlsx`.
 - :pull:`367`: Add a parameter for exporting all model+scenario run versions to :meth:`.Platform.export_timeseries_data`, and fix a bug where exporting all runs happens uninteneded.

--- a/ixmp/tests/core/test_timeseries.py
+++ b/ixmp/tests/core/test_timeseries.py
@@ -266,7 +266,11 @@ def test_get(ts, fmt):
 
     # Data can be retrieved and has the expected value
     obs = ts.timeseries(**args)
-    assert_frame_equal(exp, obs, check_like=True)
+
+    # NB this included check_like=True to be tolerant of JDBCBackend returning columns
+    #    in unpredictable order. In pandas 1.2.0, this caused an exception; see
+    #    pandas-dev/pandas#39168. Removed until the upstream bug is fixed.
+    assert_frame_equal(exp, obs)
 
 
 @pytest.mark.parametrize("fmt", ["long", "wide"])

